sklearn/preprocessing/mice.py

```diff
@@ -0,0 +1,365 @@
+        self.missing_values = missing_values
+        self.imputation_order = imputation_order
+        self.n_imputations = n_imputations
+        self.n_burn_in = n_burn_in
+        self.n_nearest_columns = n_nearest_columns
+        self.initial_fill_method = initial_fill_method
+        self.min_value = min_value
+        self.max_value = max_value
+        self.verbose = verbose
+        self.trained_model_triplets = []
+
+    def _fill_in_one_column(self,
+                            X_filled,
+                            mask_missing_values,
+                            this_column,
+                            other_columns,
+                            model=None,
+                            min_std=1e-6):
+        # if nothing is missing, just return the default
+        if mask_missing_values[:, this_column].sum() == 0:
+            return X_filled, model
+        missing_row_mask = mask_missing_values[:, this_column]
+        if model is None:
+            X_train = X_filled[:, other_columns][~missing_row_mask]
+            y_train = X_filled[:, this_column][~missing_row_mask]
+            if np.std(y_train) > 0:
+                model = deepcopy(self.model)
+                model.fit(X_train, y_train)
+            else:
+                model = DummyRegressor()
+                model.fit(X_train, y_train)
+        X_test = X_filled[:, other_columns][missing_row_mask]
+        mus, sigmas = model.predict(X_test, return_std=True)
+        sigmas = np.maximum(sigmas, min_std)
+        imputed_values = np.random.normal(mus, sigmas)
+        imputed_values = self._clip(imputed_values)
+        X_filled[missing_row_mask, this_column] = imputed_values
+        return X_filled, model
+
+    def _get_other_columns(self,
+                           n_features,
+                           this_column,
+                           abs_correlation_matrix):
+        """
+        Get a list of other columns to predict this_column.
+
+        If self.n_nearest_columns is less than or equal to the total
+        number of columns, then use a probability proportional to the absolute
+        correlation between this_column and each other column to randomly
+        choose a subsample of the other columns (without replacement).
+
+        Parameters
+        ----------
+        n_features : integer
+            Number of columns in X.
+
+        this_column : integer
+            Index of the column currently being imputed.
+
+        abs_correlation_matrix : array-like, shape (n_features, n_features)
+            Absolute correlation matrix of X at the begging of the current
+            round. The diagonal has been zeroed out and each column has been
+            normalized to sum to 1.
+        """
+        ordered_columns = np.arange(n_features)
+        if self.n_nearest_columns <= n_features - 1:
+            p = abs_correlation_matrix[:, this_column]
+            other_columns = np.random.choice(ordered_columns,
```

Comment:
I don't see a `choice` function in here? https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/fixes.py